.github/workflows/nix-rebuild.yml

```diff
@@ -1,47 +1,126 @@
+      # Calculate changed derivations and upload as artifacts
+      - run: nix eval --impure --expr 'import ./.github/workflows/changedPaths.nix ./pr_head ./pr_base' --json | tee chDerivs.json
+      # Upload JSON of the changed derivations
+      - uses: actions/upload-artifact@v3
+        with:
+          name: Changed-Derivations
+          path: chDerivs.json
+          retention-days: 1
+
+  # Rebuild changed derivations for the pinned snapshot
+  rebuild-pin:
+    name: Pinned
+    needs: get-changed-derivs
+    runs-on: ubuntu-latest
+    steps:
+      # Install Nix on the runner
+      - uses: cachix/install-nix-action@v20
+      # Pull from the cachix cache
+      - uses: cachix/cachix-action@v12
+        with:
+          name: nix-qchem
+      # Checkout of the current head in the working dir
+      - uses: actions/checkout@v3
+      # Get changed derivations as artifact from previous step
+      - uses: actions/download-artifact@v3
+        with:
+          name: Changed-Derivations
+      # Rebuild all changed python3 packages
+      - run: |
+          if [ -n "$(jq '.python3[]' chDerivs.json)" ]; then
+            nix build --accept-flake-config --print-build-logs $(for p in $(jq '.python3[]' chDerivs.json | tr -d "\""); do echo ".#python3.pkgs.${p}"; done)
+          fi
+      # Rebuild all changed top-level packages
+      - run: |
+          if [ -n "$(jq '.topLevel[]' chDerivs.json)" ]; then
+            nix build --accept-flake-config --print-build-logs $(for p in $(jq '.topLevel[]' chDerivs.json | tr -d "\""); do echo ".#${p}"; done)
+          fi
+
+  # Rebuild changed derivations for nixpkgs-unstable
+  rebuild-unstable:
+    name: Unstable
+    needs: get-changed-derivs
+    runs-on: ubuntu-latest
+    steps:
+      # Install Nix on the runner
+      - uses: cachix/install-nix-action@v20
+      # Pull from the cachix cache
+      - uses: cachix/cachix-action@v12
+        with:
+          name: nix-qchem
+      # Checkout of the current head in the working dir
+      - uses: actions/checkout@v3
+      # Get changed derivations as artifact from previous step
+      - uses: actions/download-artifact@v3
+        with:
+          name: Changed-Derivations
+      # Rebuild all changed python3 packages
+      - run: |
+          if [ -n "$(jq '.python3[]' chDerivs.json)" ]; then
+            nix build --accept-flake-config --override-input nixpkgs nixpkgs/nixpkgs-unstable --print-build-logs $(for p in $(jq '.python3[]' chDerivs.json | tr -d "\""); do echo ".#python3.pkgs.${p}"; done)
+          fi
+      # Rebuild all changed top-level packages
+      - run: |
+          if [ -n "$(jq '.topLevel[]' chDerivs.json)" ]; then
+            nix build --accept-flake-config --override-input nixpkgs nixpkgs/nixpkgs-unstable --print-build-logs $(for p in $(jq '.topLevel[]' chDerivs.json | tr -d "\""); do echo ".#${p}"; done)
+          fi
+
+  # Rebuild changed derivations for nixpkgs/master
+  rebuild-master:
+    name: Master
```

Comment:
I actually wonder if we really should build PRs against nixpkgs-master. The master branch itself contains a lot of temporary breakage and large rebuild sets, that may not have been built by Hydra. I do not even build locally against nixpkgs-master as this would result in a lot of irrelevant build failures. I think that building against nixpkgs-unstable completely covers our use-case of projecting ahead.
Would it maybe make more sense to build the merged PR against the pinned nixpkgs here?

Reply:
I agree, the master rebuild could be removed.

>  Would it maybe make more sense to build the merged PR against the pinned nixpkgs here?

What exactly do you mean? :) Not building the head of the PR but the result after the merge commit?

Reply:
> Yes, build the result of the merge. (I.e. what would happen if one were to merge a PR).

Done in https://github.com/Nix-QChem/NixOS-QChem/pull/266/commits/69038e31debd40a39d08f1e74d1557381bbf6223

In GitHub action I find this to be quite obscure ...